src/serde_bin.rs

```diff
@@ -61,12 +61,19 @@ pub trait DeBin: Sized {
 
 /// The error message when failing to deserialize from raw bytes.
 #[derive(Clone)]
+#[non_exhaustive]
```

Comment:
I'm not ready to add new error messages right now, but this will at least let us make changes to this in future so that we can include a message field like in `DeRonError` or `DeJsonError`